module/programs/helper.js

```diff
@@ -1057,19 +1057,31 @@ module.exports = class ProgramsHelper {
                 status: httpStatusCode.bad_request.status
               }
             }
+            
           }
-          pushProgramUsersDetailsToKafka = true; 
+          
+        }
+
+        // if requestForPIIConsent Is false and user not joined program till now then set pushProgramUsersDetailsToKafka = true; 
+        // if requestForPIIConsent == true and data.consentShared value is true which means user interacted with the consent popup set pushProgramUsersDetailsToKafka = true;
+        if((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === false && !programUsersDetails.length > 0) ||
+          ((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === true) && (data.hasOwnProperty('consentShared') && data.consentShared == true) ) ){
+            pushProgramUsersDetailsToKafka = true; 
         }
         
         //create or update query
         const query = { 
           programId: programId,
           userId: userId
         };
-        
+        //if a resource is started
         if ( data.isResource ) {
           programUsersData.resourcesStarted = true;
         }
+        //if user interacted with the consent-popup
+        if ( data.consentShared ) {
+          programUsersData.consentShared = true;
```

Comment:
@VISHNUDAS-tunerlabs we should return this new key in any existing API response?

Reply:
@aks30 we can pass. added the changes. also need to make changes to ml-survey will raise PR on that